libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1699,7 +1699,13 @@ void TypeChecker::typeCheckFunctionCall(
 		);
 		return;
 	}
-
+	if (_functionType->kind() == FunctionType::Kind::Internal && _functionType->hasDeclaration())
+		if (auto const* functionDefinition = dynamic_cast<FunctionDefinition const*>(&_functionType->declaration()))
+			if (functionDefinition->annotation().contract != m_scope && !functionDefinition->isImplemented())
```

Comment:
This here alone is what actually fixes the bug that triggered this PR...
And I'm not actually sure, if it's too crude :-)...

Reply:
Can you add a comment saying that `functionDefinition->annotation().contract != m_scope` ensures that this is from a "qualified" access like `A.f` instead of just `f`?

Reply:
Hm... problem is: it *didn't*...